pkg/tasks/tasks.go

```diff
@@ -110,7 +110,7 @@ type Task interface {
 	Run() error
 }
 
-type taskErr struct {
-	error
-	name string
+type TaskErr struct {
```

Comment:
I think that we can improve the API here. IMHO it would make sense to have a representation of "all errors from a task group":

```golang
type TaskGroupErrors []TaskErr

func (tge TaskGroupErrors) Error() string {
	s := make([]string, 0, len(tge))
	for _, te := range tge {
		s = append(s, fmt.Sprintf("%v: %v", te.Name, te.Err))
	}
	return strings.Join(s, ", ")
}
```

Since `TaskGroupError` implements the [error interface](https://pkg.go.dev/builtin#error), we could write `klog.Infof("ClusterOperator reconciliation failed (attempt %d), retrying. Err: %v", tgErrors)`.

Reply:
I am not convinced that we need another type just for printing the error out. I already have some code that iterates through the errors, gets the error string and creates a new error with the concatenated error messages.

Reply:
Done